### PR TITLE
Added RTCM subscriber and NMEA GGA publisher

### PR DESCRIPTION
**Pull request by cledant**
In order to support a third party NTRIP client ROS2 module, an RTCM subscriber and NMEA GGA publisher were added.

### CODE REVIEW

File: src/message_wrapper.cpp [hunk 1]
```diff
@@ -510,6 +510,33 @@ void MessageWrapper::LLtoUTM(double Lat, double Long, int zoneNumber, double &UT
   }
 }
 
+uint32_t MessageWrapper::convertSbgGpsTypeToNmeaGpsType(uint32_t sbgGpsType)
+{
+    switch (sbgGpsType)
+    {
+        case SBG_ECOM_POS_NO_SOLUTION:
```
Comment (AlexandrePTJ) [inline src/message_wrapper.cpp:517]: Redundant with default case

File: include/sbg_driver/message_wrapper.h [hunk 2]
```diff
@@ -316,7 +330,15 @@ class MessageWrapper : public rclcpp::Node
...
-   void LLtoUTM(double Lat, double Long, int zoneNumber, double &UTMNorthing, double &UTMEasting) const;
+  void LLtoUTM(double Lat, double Long, int zoneNumber, double &UTMNorthing, double &UTMEasting) const;
+
+  /*!
+   * Convert SbgEComGpsPosType enum to SbgNmeaGpsQuality enum
+   *
+   * \param[in] sbgGpsType              SbgECom GPS type
+   * \return                            NMEA GPS type
+   */
+  static uint32_t convertSbgGpsTypeToNmeaGpsType(uint32_t sbgGpsType);
```
Comment (AlexandrePTJ) [inline include/sbg_driver/message_wrapper.h:341]: Replace uint32_t with explicit enum name

File: src/message_wrapper.cpp [hunk 3]
```diff
@@ -1310,3 +1337,149 @@ const sensor_msgs::msg::FluidPressure MessageWrapper::createRosFluidPressureMess
 
   return fluid_pressure_message;
 }
+
+const nmea_msgs::msg::Sentence sbg::MessageWrapper::createSbgGpsPosMessageGGA(const SbgLogGpsPos &ref_log_gps_pos) const
+{
+    nmea_msgs::msg::Sentence gps_pos_nmea_msg;
```
Comment (AlexandrePTJ) [inline src/message_wrapper.cpp:1343]: Why not using nmea_msgs::msg:Gpgga message directly ?

File: include/sbg_driver/sbg_device.h [hunk 4]
```diff
@@ -41,6 +42,7 @@ class SbgDevice
   SbgInterface            m_sbg_interface_;
   rclcpp::Node&        	  m_ref_node_;
   MessagePublisher        m_message_publisher_;
+  std::shared_ptr<MessageSubscriber> m_message_subscriber_;
```
Comment (AlexandrePTJ) [inline include/sbg_driver/sbg_device.h:45]: Re-align other members

File: src/message_wrapper.cpp [hunk 5]
```diff
@@ -1310,3 +1337,149 @@ const sensor_msgs::msg::FluidPressure MessageWrapper::createRosFluidPressureMess
...
+    uint32_t current_hour = nb_hours % 24;
+    uint32_t current_minute = nb_minutes % 60;
+    uint32_t current_second = (ref_log_gps_pos.timeOfWeek / 1000) % 60;
+    uint32_t current_ms = ref_log_gps_pos.timeOfWeek % (60 * 1000);
+    current_ms = (current_ms - (current_second * 1000)) / 10;
+
+    // Latitude conversion
+    char lat_dir;
+    float latitude = ref_log_gps_pos.latitude;
+    if (latitude >= 1000.0f)
```
Comment (AlexandrePTJ) [inline src/message_wrapper.cpp:1357]: Latitude should be in range [-90,90]. So used value should be span in this range.

File: src/message_wrapper.cpp [hunk 6]
```diff
@@ -1310,3 +1337,149 @@ const sensor_msgs::msg::FluidPressure MessageWrapper::createRosFluidPressureMess
...
+    }
+    else
+    {
+        lat_dir = 'N';
+    }
+
+    // Longitude conversion
+    char lon_dir;
+    float longitude =  ref_log_gps_pos.longitude;
+    if (longitude >= 10000.0f)
```
Comment (AlexandrePTJ) [inline src/message_wrapper.cpp:1381]: Longitude should be in range [-180,180]. So used value should be span in this range.

File: src/message_wrapper.cpp [hunk 7]
```diff
@@ -1310,3 +1337,149 @@ const sensor_msgs::msg::FluidPressure MessageWrapper::createRosFluidPressureMess
...
+        lon_mins *= -1.0f;
+        lon_dir = 'W';
+    }
+    else
+    {
+        lon_dir = 'E';
+    }
+
+    // DOP computation
+    double h_dop = sqrt((ref_log_gps_pos.latitudeAccuracy * ref_log_gps_pos.latitudeAccuracy) + \
```
Comment (AlexandrePTJ) [inline src/message_wrapper.cpp:1403]:
```suggestion
    double h_dop = std::hypot(ref_log_gps_pos.latitudeAccuracy, ref_log_gps_pos.longitudeAccuracy);
```

**Review by AlexandrePTJ — COMMENTED** [review]
Needs some changes and some clarifications

File: src/message_wrapper.cpp (hunk 3 above)
Comment (cledant) [reply src/message_wrapper.cpp:1343]: Package nmea_msgs only provides msg type.  
There is no serialization / deserialization provided.  
Thus if a third party package handling ntrip acts as passthrough, it needs to be already serialized.

File: src/message_wrapper.cpp (hunk 7 above)
Comment (cledant) [reply src/message_wrapper.cpp]: Done

File: src/message_wrapper.cpp (hunk 6 above)
Comment (cledant) [reply src/message_wrapper.cpp]: Done

File: include/sbg_driver/message_wrapper.h (hunk 2 above)
Comment (cledant) [reply include/sbg_driver/message_wrapper.h]: Done

File: src/message_wrapper.cpp (hunk 1 above)
Comment (cledant) [reply src/message_wrapper.cpp]: Done

**Pull request merged**